### PR TITLE
[SIG 4412] Enable to make category publicly available

### PR DESCRIPTION
**Pull request by kiruys**
- Checkbox and textfield for is_public_available and public_name respectively in CategoryForm
- Possible values for both variables are retrieved from the data and stored in the CategoryForm state
- New values for both values can be submitted and retrieved
- Tests for both fields

### CODE REVIEW

File: src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx [hunk 1]
```diff
@@ -139,7 +162,42 @@ const CategoryForm: FunctionComponent<CategoryFormProps> = ({
             ) : null}
 
             <FieldGroup>
-              <Label>Afhandeltermijn</Label>
+              <StyledDefinitionTerm>
+                <strong>Openbaar tonen</strong>
+              </StyledDefinitionTerm>
```
Comment (janjaap) [inline src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx:167]: Every `dt` should have a corresponding `dd`.

File: src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx [hunk 2]
```diff
@@ -95,6 +97,27 @@ const CategoryForm: FunctionComponent<CategoryFormProps> = ({
...
+    }
+  }, [
+    data?.is_public_accessible,
+    data?.public_name,
+    setIsPublicAccessible,
+    setPublicName,
+  ])
+
+  const onCheck = useCallback(
+    (evt) => {
```
Comment (janjaap) [inline src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx:115]: Can you add a type to this parameter?

File: src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx [hunk 3]
```diff
@@ -95,6 +97,27 @@ const CategoryForm: FunctionComponent<CategoryFormProps> = ({
     [data]
   )
 
+  useEffect(() => {
+    if (data?.is_public_accessible) {
+      setIsPublicAccessible(data.is_public_accessible)
+    }
```
Comment (janjaap) [inline src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx:103]: I think you can skip this bit when line 88 is this:
```javascript
const [isPublicAccessible, setIsPublicAccessible] = useState(data.is_public_accessible ?? false)
```

File: src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx [hunk 4]
```diff
@@ -95,6 +97,27 @@ const CategoryForm: FunctionComponent<CategoryFormProps> = ({
...
+    if (data?.public_name) {
+      setPublicName(data.public_name)
+    }
```
Comment (janjaap) [inline src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx:106]: Same here; the state value can be set on component instantiation.

File: src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx (hunk 3 above)
Comment (kiruys) [reply src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx]: done

File: src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx (hunk 4 above)
Comment (kiruys) [reply src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx]: done

File: src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx (hunk 2 above)
Comment (kiruys) [reply src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx:102]: done

File: src/signals/settings/categories/Detail/index.js [hunk 5]
```diff
@@ -96,7 +97,12 @@ const CategoryDetail = () => {
     (event) => {
       const formData = [...new FormData(event.target.form).entries()]
         // convert stringified boolean values to actual booleans
-        .map(([key, val]) => [key, key === 'is_active' ? val === 'true' : val])
+        .map(([key, val]) => [
+          key,
+          key === 'is_active' || key === 'is_public_accessible'
```
Comment (janjaap) [inline src/signals/settings/categories/Detail/index.js:102]: Can you add a test that covers the branch?

Comment (kiruys) [reply src/signals/settings/categories/Detail/index.js:102]: I did

File: src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx (hunk 1 above)
Comment (janjaap) [reply src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx:159]: I don't see the element in the DOM

Comment (kiruys) [reply src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx:159]: maybe you can look in the code
Schermafbeelding 2022-03-08 om 11 11 02

Comment (kiruys) [reply src/signals/settings/categories/Detail/components/CategoryForm/CategoryForm.tsx]: I have replaced the headings

**Pull request merged**